### PR TITLE
Don't commit kafka topic on failures

### PR DESCRIPTION
**Pull request by filosganga**
This PR disable the Kafka auto-commit.

It also refactors the Kafka package to be named ingester and fix the service tests.

### CODE REVIEW

File: src/main/scala/com/ovoenergy/comms/profiles/ingester/UserEvent.scala [hunk 1]
```diff
@@ -80,7 +80,8 @@ case class UserEvent(
     emailAddress = user.email.map(EmailAddress),
     phoneNumber = user.phoneNumbers
       .map(PhoneNumberValidation.validateAndFormatNumber(_).map(PhoneNumber))
-      .head,
+      .headOption
+      .flatten,
     communicationPreferences = user.marketingPreferences
       .map(p => CommunicationPreference(Marketing, List(Channel.unsafeFromString(p)))), // TODO: align this with identities model
```
Comment (laurence-bird) [inline src/main/scala/com/ovoenergy/comms/profiles/ingester/UserEvent.scala]: I guess we should verify this before we deploy to PRD with the identity team, or deploy to UAT and test it out? I would assume the source of this data is the same served from SF but would be good to verify

File: src/main/scala/com/ovoenergy/comms/profiles/Main.scala [hunk 2]
```diff
@@ -33,13 +33,21 @@ object Main extends StreamApp[IO] with AvroCodec {
...
+              customerRepository.storeCustomer(event.toCustomer).onError {
+                case e => IO(log.error("Failed to write to customer repository", e))
+              }
           }
 
         val identityDeleteStream: Stream[IO, Unit] =
-          KafkaConsumer[IO](_.deleteTopic) { (userDeleteEvent: Record[UserDeletedEvent]) =>
-            customerRepository.deleteCustomer(CustomerId(Realm.Ovo, userDeleteEvent.value.id))
+          KafkaConsumer[IO](config.ingester, _.deleteTopic) { userDeleteEvent: Record[UserDeletedEvent] =>
+            val event = userDeleteEvent.value()
```
Comment (laurence-bird) [inline src/main/scala/com/ovoenergy/comms/profiles/Main.scala]: probs shouldn't be logging the entire event in hindsight

File: src/main/scala/com/ovoenergy/comms/profiles/ingester/UserEvent.scala (hunk 1 above)
Comment (filosganga) [reply src/main/scala/com/ovoenergy/comms/profiles/ingester/UserEvent.scala]: Even if they tell us that the list will never been empty, I would prefer to be more defensive.

Comment (filosganga) [reply src/main/scala/com/ovoenergy/comms/profiles/ingester/UserEvent.scala]: For SF we were doing this:

```scala
  def findValidUkMobileNumber(telephoneNumbers: TelephoneNumbers): Option[String] = {
    telephoneNumbers.mobile
      .flatMap(validateAndFormatNumber)
      .orElse(telephoneNumbers.home.flatMap(validateAndFormatNumber))
      .orElse(telephoneNumbers.work.flatMap(validateAndFormatNumber))
  }
```

I believe we should do the same here.

**Pull request merged**